### DISCUSSION TITLE
What is the difference between "direction" and "angles" pars in vario_estimate() function?

### DISCUSSION THREAD

**Discussion by flydream0428** [GeoStat-Framework/GSTools#266] [Q&A] [answered]
I have tried to estimate a directional variogram by the 'direction' and 'angles' parameters. It turns out it gives very similar results. 
For 'direction', I am using, 
```python
bin_center, dir_vario_ani = gs.vario_estimate(
    pos=(x_con, y_con), 
    field=z_con, 
    direction=gs.rotated_main_axes(dim=2, angles=1/8*np.pi),
)
```
For 'angles', I am using, 
```python
bin_center, dir_vario_ani = gs.vario_estimate(
    pos=(x_con, y_con), 
    field=z_con, 
    angles=[1/8*np.pi, 5/8*np.pi],
)
```

What is the right one to use here? I have tried to read the function docstring, but I am still confused.

**Comment by MuellerSeb** [comment, marked as answer]
Hey there!

This is a very good question. You can use both to get the same results. In case of `direction`, you need to pass a vector (or multiple vectors) that points in the direction you want to estimate the variogram.

If you use `angles` you can describe the same directions, but in [Spherical coordinates](https://en.wikipedia.org/wiki/N-sphere#Spherical_coordinates).

In your case
```python
gs.rotated_main_axes(dim=2, angles=1/8*np.pi))
```
Gives two direction vectors:
1. pointing in the direction with an angle of 22,5°
2. the orthogonal direction to the first direction (112,5° or `5/8*np.pi`)

This means
```python
direction=gs.rotated_main_axes(dim=2, angles=1/8*np.pi)
# and
angles=[1/8*np.pi, 5/8*np.pi]
```
are describing the same directions. 

Only thing to take care about is, that the `angles` parameter in `rotated_main_axes` describes the rotation angles of a covariance model (like [Tait–Bryan](https://en.wikipedia.org/wiki/Euler_angles#Tait-Bryan_angles) angles in 3D) where you need `d * (d - 1) / 2` angles (1D: 0, 2D: 1, 3D: 3, 4D: 6, ...) to fully describe rotation, whereas the `angles` parameter in `vario_estimate` are the mentioned spherical coordinates, where you only need `(d - 1)` angles (1D: 0, 2D: 1, 3D: 2, 4D: 3, ...).

In two dimension the number and the meaning of these angles coincide, therefore everything worked fine here. See the [example](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/examples/01_random_field/07_higher_dimensions.html) about higher dimension to get an explanation of the rotation angles of a covariance model.

Hope that helps.
Sebastian
